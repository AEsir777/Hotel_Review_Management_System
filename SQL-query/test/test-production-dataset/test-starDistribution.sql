/* test star distribution */
CALL getStarDistribution('"--7PUidqRWpRSpXebiyxTg"');
CALL getStarDistribution('"-7KAng29RoHr87mvOFbK9w"');
CALL getStarDistribution('"-7wXZWnKe26iwJY_5z_AAw"');


/* update review */
CALL UpdateReviewTextByRid('"3Ta-cyj4Iu8XBAltOrdj6A"',
 "update", 1);
/* test if stars gets updated in Business*/
CALL getStarDistribution('"--7PUidqRWpRSpXebiyxTg"');




/* not exist */
CALL getStarDistribution('B100');
CALL getStarDistribution(NULL);