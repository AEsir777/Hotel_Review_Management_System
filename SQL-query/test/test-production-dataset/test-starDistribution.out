/* test star distribution */
// CALL getStarDistribution("--7PUidqRWpRSpXebiyxTg");
5_star	4_star	3_star	2_star	1_star	0_star
0	0	1	0	0	0
// expected:
// 5_star	4_star	3_star	2_star	1_star	0_star
// 0	0	1	0	0	0





// CALL getStarDistribution("-7KAng29RoHr87mvOFbK9w");
# 5_star, 4_star, 3_star, 2_star, 1_star, 0_star
'0', '0', '1', '0', '0', '0'
// expected output:
// # 5_star, 4_star, 3_star, 2_star, 1_star, 0_star
// '0', '0', '1', '0', '0', '0'





// CALL getStarDistribution("-7wXZWnKe26iwJY_5z_AAw");
# 5_star, 4_star, 3_star, 2_star, 1_star, 0_star
1, 0, 0, 0, 0, 0
// expected output:
// # 5_star, 4_star, 3_star, 2_star, 1_star, 0_star
// 1, 0, 0, 0, 0, 0



/* update review */
// CALL UpdateReviewTextByRid("3Ta-cyj4Iu8XBAltOrdj6A", "update", 1);
// CALL getStarDistribution("--7PUidqRWpRSpXebiyxTg");
# 5_star, 4_star, 3_star, 2_star, 1_star, 0_star
0, 0, 0, 0, 1, 0
// expected output:
// # 5_star, 4_star, 3_star, 2_star, 1_star, 0_star
// 0, 0, 0, 0, 1, 0





/* not exist */
// CALL getStarDistribution('B100');
NULL
// expected:
// NULL

// CALL getStarDistribution(NULL);
NULL
// expected:
// NULL